### PR TITLE
Issue #378 by pjohans: allow other modules (ting_infomedia) to alter …

### PR DESCRIPTION
**Pull request by pjohans**
…infomedia url

### CODE REVIEW

File: ting.entities.inc [hunk 1]
```diff
@@ -316,6 +317,10 @@ class TingEntity extends DingEntity {
       drupal_alter('ting_online_url', $url, $this);
     }
 
+    if(strpos($url, '[useraccessinfomedia]') === 0){
+      drupal_alter('infomedia_url', $url, $this);
+    }
```
Comment (kasperg) [inline ting.entities.inc]: I do not think the right way to go about this is to introduce a new hook when we already have `ting_online_url`.

As I see it the problem is that infomedia urls are not identified as proper  urls in line 303 as they do not begin with 'http'. In that case I would much prefer that the detection code is updated to support urls with a leading token instead of one-off solutions.

File: ting.client.inc [hunk 2]
```diff
@@ -687,8 +688,10 @@ function ting_execute($request) {
     }
 
     return $response;
-  }
-  catch (TingClientException $e) {
+  } catch (TingClientException $e) {
+    if (isset($e->user_message)) {
+      drupal_set_message($e->user_message, 'warning');
+    }
```
Comment (kasperg) [inline ting.client.inc]: This change does not seem to be related to the issue at hand.

**Comment by kasperg** [comment]
Review complete. I have one comment and then an unrelated commit has snuck in.

**Comment by JrgenGNielsen** [comment]
@kasperg New pull request: [https://github.com/ding2/ting/pull/40]
